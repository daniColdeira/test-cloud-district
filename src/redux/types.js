//Funcionamiento
export const CHECK_LOGIN = "CHECK_LOGIN";
export const LOG_OUT = "LOG_OUT";
//-----------------------------------------
//Users
export const GET_USERS_FAILURE = "GET_USERS_FAILURE";
export const GET_USERS_SUCCESS = "GET_USERS_SUCCESS";

export const GET_USER_SUCCESS = "GET_USER_SUCCESS";
export const GET_USER_FAILURE = "GET_USER_FAILURE";
export const GET_USER_REQUEST = "GET_USER_REQUEST";

export const SET_LOADING ="SET_LOADING";

export const SET_USER_SUCCESS = "SET_USER_SUCCESS";
export const SET_USER_FAILURE = "SET_USER_FAILURE";
export const SET_USER_REQUEST = "SET_USER_REQUEST";