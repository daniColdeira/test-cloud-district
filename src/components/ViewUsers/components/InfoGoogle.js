import React, {useContext} from 'react';
import { store } from '../../../store/store.js';
import Logout from '../../GoogleLogout';
import { InfoGoogle, DivGoogle } from '../styled';

function Info() {
    const globalState = useContext(store);
    
    return (
        <InfoGoogle>
            <img src={globalState.state.user.profileObj && globalState.state.user.profileObj.imageUrl} alt="logo" />
            <DivGoogle>
                <div>
                    {globalState.state.user.profileObj && globalState.state.user.profileObj.name}
                </div>
                <div>
                    {globalState.state.user.profileObj && globalState.state.user.profileObj.email}
                </div>
            </DivGoogle>
            <Logout />
        </InfoGoogle>
    );
}

export default Info;