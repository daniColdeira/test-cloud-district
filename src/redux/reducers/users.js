import { 
  GET_USERS_SUCCESS, 
  GET_USERS_FAILURE, 
  GET_USER_SUCCESS, 
  GET_USER_FAILURE,
  GET_USER_REQUEST,  
  SET_USER_REQUEST, 
  SET_USER_SUCCESS,
  SET_USER_FAILURE
} from "../types.js";

const initialState = {
  users: [],
  user: undefined,
  response: undefined,
  error: undefined,
  loadingUser: false,
  loadingResponse: false,
};

const users = (previousState = initialState, action) => {
  switch (action.type) {
    case GET_USERS_SUCCESS:
      return Object.assign({}, previousState, { users: action.payload });
    case GET_USERS_FAILURE:
      return Object.assign({}, previousState, { error: action.payload });
    case GET_USER_SUCCESS:
      return Object.assign({}, previousState, { user: action.payload }); // Si existe Token, nos envía a la comprobación del <Redirect />
    case GET_USER_FAILURE:
      return Object.assign({}, previousState, { error: action.payload });
    case GET_USER_REQUEST:
      return Object.assign({}, previousState, { loadingUser: action.payload }); // Si existe Token, nos envía a la comprobación del <Redirect />
    case SET_USER_SUCCESS:
      return Object.assign({}, previousState, { response: action.payload });
    case SET_USER_FAILURE:
      return Object.assign({}, previousState, { error: action.payload });
    case SET_USER_REQUEST:
      return Object.assign({}, previousState, { loadingResponse: action.payload });
    default:
      return previousState;
  }
};

export default users;
