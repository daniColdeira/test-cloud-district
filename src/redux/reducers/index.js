import { combineReducers } from "redux";
import users from "./users";

const appReducer = combineReducers({
  users,
});

const rootReducer = () => (state, action) => {
  if (action.type === 'LOG_OUT') {
    state = undefined;
  }
  return appReducer(state, action);
};

export default rootReducer;
