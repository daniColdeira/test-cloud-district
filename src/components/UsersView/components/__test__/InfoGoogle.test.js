import React from 'react';
import { render } from '@testing-library/react';
import InfoGoogle from '../InfoGoogle';
import renderer from 'react-test-renderer';
import Context, {AuthContext} from '../../../../AuthContext/AuthContext.js';
import TestRenderer from 'react-test-renderer';
import { Provider } from "react-redux";
import reducers from '../../../../redux/reducers'
import { createStore, applyMiddleware } from "redux";


import { Router } from "react-router-dom"; //enrutamiento
import historyRouting from "../../../../history"; //historial de rutas

const store = createStore(
    reducers,
  );
test("renders correctly", () => {
    const tree = renderer
    .create(
        <Context>
            <Provider store={store}>
                <Router history={historyRouting}>
                    <InfoGoogle />
                </Router>
            </Provider>
            
        </Context>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});



test("render with user", () => {
    let setUserAuth = jest.fn();
    let user = { user: { profileObj:{ name:'Daniel Coldeira', email:'danielcoldeirarm@gmail.com'}}}
    const element = new TestRenderer.create(
        <AuthContext.Provider value={{ user, setUserAuth }}>
            <Provider store={store}>
                <InfoGoogle />
            </Provider>
        </AuthContext.Provider>
    );
    expect(element).toMatchSnapshot();
});