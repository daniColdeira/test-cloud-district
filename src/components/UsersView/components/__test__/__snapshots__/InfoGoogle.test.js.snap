// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with user 1`] = `
<div
  className="sc-hBMVcZ gJpLAz"
>
  <img
    alt="logo"
  />
  <div
    className="sc-fKgIGh kKkNGW"
  >
    <div>
      Daniel Coldeira
    </div>
    <div>
      danielcoldeirarm@gmail.com
    </div>
  </div>
  <button
    className="sc-eCApGN kBtpMM"
    onClick={[Function]}
  >
    <span>
      LogOut
    </span>
  </button>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="sc-hBMVcZ gJpLAz"
>
  <img
    alt="logo"
  />
  <div
    className="sc-fKgIGh kKkNGW"
  >
    <div />
    <div />
  </div>
  <button
    className="sc-eCApGN kBtpMM"
    onClick={[Function]}
  >
    <span>
      LogOut
    </span>
  </button>
</div>
`;
